kubernetes/filters.go

```diff
@@ -58,3 +58,13 @@ func FilterPodsForController(controllerName string, controllerType string, allPo
 	}
 	return pods
 }
+
+func FilterServicesForSelector(selector labels.Selector, allServices []v1.Service) []v1.Service {
+	var services []v1.Service
+	for _, svc := range allServices {
+		if selector.Matches(labels.Set(svc.Spec.Selector)) {
```

Comment:
I checked how openshift is grouping services into apps, and it seems to take the selector as you do, and not the service labels, so I'm finally with you here to be consistent with openshift (it took me some time to admit :-) )